src/cmd/sanity_test/sanity_tests.go

```diff
@@ -112,6 +119,76 @@ func checkEmailRestoration(client *msgraphsdk.GraphServiceClient, testUser, fold
 				*restore.GetTotalItemCount())
 			os.Exit(1)
 		}
+
+		checkAllSubFolder(user, restore, "", messageCount)
+	}
+}
+
+func getAllSubFolder(
+	user *users.UserItemRequestBuilder,
+	r models.MailFolderable,
+	parentFolder string,
+	messageCount map[string]int32,
+) {
+	folder := user.MailFoldersById(*r.GetId())
+
+	childFolder, err := folder.ChildFolders().Get(context.Background(), nil)
+	if err != nil {
+		fmt.Printf("Error getting the drive: %v\n", err)
+		os.Exit(1)
```

Comment:
Can this get rebased onto my other changes so that we can call `fatal()` here? (and minimize rebase collisions)